SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorModule.cpp

```diff
@@ -230,6 +234,45 @@ bool FSpatialGDKEditorModule::ForEveryServerWorker(TFunction<void(const FName&,
 	return false;
 }
 
+FPlayInEditorSettingsOverride FSpatialGDKEditorModule::GetPlayInEditorSettingsOverrideForTesting(UWorld* World) const
+{
+	FPlayInEditorSettingsOverride PIESettingsOverride = ISpatialGDKEditorModule::GetPlayInEditorSettingsOverrideForTesting(World);
+	if (const ASpatialWorldSettings* SpatialWorldSettings = Cast<ASpatialWorldSettings>(World->GetWorldSettings()))
+	{
+		EMapTestingMode TestingMode = SpatialWorldSettings->TestingSettings.TestingMode;
+		if(TestingMode != EMapTestingMode::UseCurrentSettings)
+		{
+			TActorIterator<ASpatialFunctionalTest> It(World);
+			if (TestingMode == EMapTestingMode::Detect)
+			{
+				TestingMode = It ? EMapTestingMode::ForceSpatial : EMapTestingMode::ForceNative;
+			}
+
+			int NumberOfClients = 1;
+
+			switch (TestingMode)
+			{
+			case EMapTestingMode::ForceNative:
+				PIESettingsOverride.bUseSpatial = false;
+				break;
+			case EMapTestingMode::ForceSpatial:
+				PIESettingsOverride.bUseSpatial = true;
+				for (; It; ++It)
+				{
+					NumberOfClients = FMath::Max(It->GetNumRequiredClients(), NumberOfClients);
```

Comment:
What's the expectations here if there are multiple ASpatialFunctionalTests? Are all the tests run in serial?

Reply:
All tests in the same map run one after another, so in this case we make sure that there's as many clients as needed by any of the tests.